projects/ion/src/lib/steps/steps.component.ts

```diff
@@ -0,0 +1,87 @@
+import { CommonModule } from '@angular/common';
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  OnInit,
+  Output,
+  SimpleChanges,
+} from '@angular/core';
+import { IonIconComponent } from '../icon';
+import { IonStepsProps, Status, StatusType, Step } from './types';
+
+@Component({
+  standalone: true,
+  selector: 'ion-steps',
+  imports: [CommonModule, IonIconComponent],
+  templateUrl: './steps.component.html',
+  styleUrls: ['./steps.component.scss'],
+})
+export class IonStepsComponent implements OnInit, OnChanges {
+  @Input() current: IonStepsProps['current'] = 1;
+  @Input({ required: true }) steps!: IonStepsProps['steps'];
+  @Input() disabled: IonStepsProps['disabled'] = false;
+  @Input() clickable: IonStepsProps['clickable'] = false;
+  @Output() indexChange: IonStepsProps['indexChange'] =
+    new EventEmitter<number>();
+
+  private firstCatchStatus = true;
+  public FIRST_STEP = 1;
+
+  public handleClick(index: number | undefined): void {
```

Comment:
you should consider organizing the methods, grouping by public, private and life cycle methods.